testing/python/cuda/test_tilelang_tma_descriptor_elem_offset.py

```
@@ -0,0 +1,168 @@
+        assert ".data_ptr() +" in python_init
+
+
+@pytest.mark.parametrize("elem_offset", [1, "symbolic"])
+def test_auto_copy_falls_back_for_unproven_descriptor_alignment(elem_offset):
+    offsets, descriptors = _descriptor_base_byte_offsets(_auto_copy_program(elem_offset), "sm_90")
+    assert offsets == []
+    assert descriptors == []
+
+
+def test_fp4_unpacked_copy_rejects_16_byte_offset():
+    with pytest.raises(Exception, match="T.tma_copy"):
```

Ruff (0.16.0)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)